### DISCUSSION TITLE
Feuille de route - Automne/Hiver 2024

### DISCUSSION THREAD

**Discussion by mguihal** [assemblee-virtuelle/archipelago#198] [Ideas]
Suite à la dernière résidence de l'Assemblée Virtuelle où les orientations stratégiques d'Archipelago ont été discutées, je propose la feuille de route court-terme suivante pour les semaines/mois à venir : 

- [Intégration d'améliorations faites pour l'instance Archipelago "Transiscope Nantes" au repo général pour qu'elles soient disponibles à tout le monde](https://github.com/assemblee-virtuelle/archipelago/milestone/3)
  - ✅  **[middleware]** Anonymiser les identifiants des utilisateurs ([#212](https://github.com/assemblee-virtuelle/archipelago/pull/212)) 🚀 [v2.4.0](https://github.com/assemblee-virtuelle/archipelago/releases/tag/v2.4.0)
  - ✅ **[middleware]** Ajouter la possibilité de modifier les permissions par défaut des containers via un fichier de configuration ([#222](https://github.com/assemblee-virtuelle/archipelago/pull/222)) 🚀 [v2.5.0](https://github.com/assemblee-virtuelle/archipelago/releases/tag/v2.5.0)
  - ✅  **[middleware]** Ajouter la possibilité de ne pas exposer les données personnelles des utilisateurs ([#212](https://github.com/assemblee-virtuelle/archipelago/pull/212)) 🚀 [v2.4.0](https://github.com/assemblee-virtuelle/archipelago/releases/tag/v2.4.0)
  - ✅ **[middleware]** Ajouter la possibilité d'avoir un utilisateur super-administrateur via un fichier de configuration ([#220](https://github.com/assemblee-virtuelle/archipelago/pull/220)) 🚀 [v2.5.0](https://github.com/assemblee-virtuelle/archipelago/releases/tag/v2.5.0)

  - ✅  **[frontend]** Changer la façon de requêter les données pour passer via LDP au lieu de SPARQL ([#199](https://github.com/assemblee-virtuelle/archipelago/issues/199)) 🚀 [v2.4.0](https://github.com/assemblee-virtuelle/archipelago/releases/tag/v2.4.0)
  - ✅ **[frontend]** Ajouter un deuxième layout et permettre de choisir/personnaliser le layout via un fichier de configuration ([#200](https://github.com/assemblee-virtuelle/archipelago/issues/200)) 🚀 [v2.3.0](https://github.com/assemblee-virtuelle/archipelago/releases/tag/v2.3.0)
  - ~~**[frontend]** Améliorer le layout actuel pour rendre la barre latérale des filtres visible en format mobile~~
  - **[frontend]** Ajouter la possibilité de sélectionner quelles ressources afficher ou non via un fichier de configuration
  - ✅   **[frontend]** Pouvoir personnaliser un élément de configuration d'une ressource via un fichier de configuration ([#226](https://github.com/assemblee-virtuelle/archipelago/pull/226)) 🚀 [v2.6.0](https://github.com/assemblee-virtuelle/archipelago/releases/tag/v2.6.0)

- [Finir la conversion des fichiers en Typescript](https://github.com/assemblee-virtuelle/archipelago/milestone/4)
- ✅  [Faire la mise-à-jour à React-admin v5](https://github.com/assemblee-virtuelle/archipelago/milestone/5) ([#224](https://github.com/assemblee-virtuelle/archipelago/pull/224))
- ✅  Simplifier la gestion des formulaires de création de ressources 🚀 [v2.9.0](https://github.com/assemblee-virtuelle/archipelago/releases/tag/v2.9.0)

---

Cette liste est sujette à débat/discussion, et n'hésitez pas à émettre vos objections en commentaires si-dessous, ou si vous voyez des sujets plus prioritaires à rajouter à la liste. Cette liste n'étant qu'une feuille de route pour se donner un cap et connaître les prochaines itérations du produit, rien n'empêche d'autres sujets d'être développés en parallèle 🙏

Comme toute roadmap, il est possible de la faire évoluer aussi en cours de route en fonction des urgences de chacun.

Chacun de ces points fera l'objet d'une création de ticket avant l'implémentation pour discuter de la mise-en-oeuvre opérationnelle de ces sujets pour discuter/débattre plus précisément à leur sujet si nécessaire.

**Comment by GuillaumeAV** [comment]
Super cette roadmap !

**Comment by fluidlog** [comment]
J'ajouterais côté frontend, la possibilité de créer une ressource (orga, projet, evenements...), en un seul formulaire, et non en deux temps comme on le fait actuellement (le titre et ensuite le reste). 
C'est ce que j'avais fait sur Carto4CH en prenant exemple sur divers projets de @srosset81 (oasis, bienvenu chez moi...).

**Reply by mguihal** [reply]
Je suis d'accord. C'est déjà le cas actuellement pour les évènements, mais c'est la seule ressource qui a été changé dans ce nouveau format pour l'instant

**Comment by fluidlog** [comment]
Commencer à réfléchir à un site Archipelago, extérieur au site AV, comme nous en avons un pour Semapps.org, déjà, pour expliquer ce que c'est.
A moins que cette page du site AV suffise : 
https://virtual-assembly.org/projets/applications/https%3A%2F%2Fdata.virtual-assembly.org%2Fprojects%2Farchipelago/

**Comment by fluidlog** [comment]
J'ajoute dans la roadmap le fait de pouvoir configurer le comportement de l'input. J'en ai besoin pour le Low-tech Lab.
Personnellement, je ne souhaite pas avoir le message "Tapez au moins 2 lettres", qui me fait fuir dès que je le lis tellement c'est affreux côté UX (sachant qu'on ne peut pas savoir quelles lettres taper pour espérer trouver des choses...).
Voir https://github.com/assemblee-virtuelle/archipelago/issues/219

**Reply by mguihal** [reply]
Ca n'a pas forcément sa place dans cette roadmap décidé au début de l'automne et qui va bientôt se finir, surtout sans discussion préalable.
Ca me semble être un petit sujet, qui peut être implémenté comme les autres PR que tu as fait non ?

**Reply by fluidlog** [reply]
Ok, je comprends qu'on ne rallonge pas cette roadmap, tu as raison, allons jusqu'au bout, du moment que je peux proposer à côté des chose :)

**Comment by mguihal** [comment]
Je pense que je vais supprimer `[frontend] Améliorer le layout actuel pour rendre la barre latérale des filtres visible en format mobile` de la roadmap actuelle.

Je ne compte personnellement pas innover/améliorer sur le layout "leftMenu" autrement que de le maintenir un minimum fonctionnel dans l'état actuel.
